private flags

sdomino opened the pull request:
This is specifically designed for spf13/cobra; a corresponding pull request from that project (https://github.com/spf13/cobra/pull/144), would allow for the addition of private flags to a project, that would run as normal however be hidden from help/usage text.

(see spf13/cobra pull request for a more detail explanation as to why this is more desirable than just using the 'Flag.Deprecated' field)

eparis commented on flag.go:
```diff
@@ -321,6 +322,18 @@ func (f *FlagSet) MarkShorthandDeprecated(name string, usageMessage string) erro
+// MarkPrivate indicated that a flag is private in your program. It will
+// continue to function but will not show up in help or usage messages. Using
+// this flag will also print the given usageMessage.
+func (f *FlagSet) MarkPrivate(name string, usageMessage string) error {
```
What is `UsageMessage` ? Looks like needless copying from the deprecated stuff?

sdomino replied:
@eparis; yes thank you, that was some c/p left-over from the MarkDeprecated method. I've removed that, and updated the comment as well.

eparis commented:
These should be suppressed in `func (f *FlagSet) FlagUsages() string {` as well? I know it's bikeshedding, but I kinda like `hidden` instead of `private`. But that's not a make or break thing. I'd also like to see some flag_test.go cases to check that hidden flags work and that they don't get printed in the usage messages.

sdomino commented:
@eparis, I'm glad you brought up the "hidden" vs "private" thing. I was on the fence about the name initially, but I do think that "hidden" better describes the functionality.

I've removed hidden flags from FlagUsages. I've added 2 tests to ensure that the flag doesn't show up in usage text, and that the flag usage message doesn't show up. I've also added "hidden" usage information to the README.md with an example.

eparis commented:
LGTM I will merge if you will squash into a single commit.  Thank you so much!
